File: Programming-Basics-With-JavaScript/Homeworks-And-Labs/05.While-Loop/02-exercise/03.vacation.js
function vacation(input) {

    let neededMoney = Number(input[0]);
    let availableMoney = Number(input[1]);
    let index = 2;
    let spendOrSave = input[index++];

    let daysCounter = 0;
    let counterSpendDays = 0;

    while (availableMoney < neededMoney) {
        daysCounter++;
        if (spendOrSave === "spend") {
            let spend = Number(input[index++]);
            availableMoney -= spend;
            if (availableMoney < 0) {
                availableMoney = 0;
            }
            counterSpendDays++;
            if (counterSpendDays === 5) {
                console.log("You can't save the money.");
                console.log(daysCounter);
                break;
            }
        } else if (spendOrSave === "save") {
            let save = Number(input[index++]);
            availableMoney += save;
            counterSpendDays = 0;
        }

        spendOrSave = input[index++];
    }

    if (availableMoney >= neededMoney) {
        console.log(`You saved the money for ${daysCounter} days.`);
    }
}
vacation(["250",
    "150",
    "spend",
    "50",
    "spend",
    "50",
    "save",
    "100",
    "save",
    "100"]);

File: Python_Basics/Homeworks-And-Labs/05.while_loop/02.exercise/03.vacation.py
needed_money = float(input())
available_money = float(input())

days_counter = 0
counter_spend_days = 0

while available_money < needed_money:
    spend_or_save = input()
    days_counter += 1
    if spend_or_save == "spend":
        spend = float(input())
        available_money -= spend
        if available_money < 0:
            available_money = 0

        counter_spend_days += 1
        if counter_spend_days == 5:
            print("You can't save the money.")
            print(days_counter)
            break
    elif spend_or_save == "save":
        save = float(input())
        available_money += save
        counter_spend_days = 0

if available_money >= needed_money:
    print(f'You saved the money for {days_counter} days.')
